packages/jazz-tools/src/tools/coValues/deepLoading.ts

```diff
@@ -6,16 +6,23 @@ import { type CoKeys } from "./coMap.js";
 import { type CoValue, type ID } from "./interfaces.js";
 
 /**
- * Used to check if T is a union type.
+ * Returns a boolean for whether the given type is a union.
  *
- * If T is a union type, the left hand side of the extends becomes a union of function types.
- * The right hand side is always a single function type.
+ * Taken from https://github.com/sindresorhus/type-fest/blob/main/source/is-union.d.ts
  */
-type IsUnion<T, U = T> = (T extends any ? (x: T) => void : never) extends (
-  x: U,
-) => void
-  ? false
-  : true;
+type IsUnion<T, U = T> = (
```

Comment:
Previous `IsUnion` was failing with `strict: true` in tsconfig. Specifically, `strictFunctionTypes`

Reply:
Nice catch!